dot/state/epoch.go

```diff
@@ -440,50 +452,105 @@ func (s *EpochState) GetLatestConfigData() (*types.ConfigData, error) {
 }
 
 // GetStartSlotForEpoch returns the first slot in the given epoch.
-// If 0 is passed as the epoch, it returns the start slot for the current epoch.
-func (s *EpochState) GetStartSlotForEpoch(epoch uint64) (uint64, error) {
-	firstSlot, err := s.baseState.loadFirstSlot()
-	if err != nil {
-		return 0, err
+func (s *EpochState) GetStartSlotForEpoch(epoch uint64, bestBlockHash common.Hash) (uint64, error) {
+	chainFirstSlotNumber, err := s.retrieveFirstNonOriginBlockSlot(bestBlockHash)
+	if err != nil && !errors.Is(err, ErrNoFirstNonOriginBlock) {
+		return 0, fmt.Errorf("retrieving first non origin block slot: %w", err)
+	}
+
+	if errors.Is(err, ErrNoFirstNonOriginBlock) && epoch == 0 {
+		slotDuration, err := s.GetSlotDuration()
+		if err != nil {
+			return 0, fmt.Errorf("getting slot duration: %w", err)
+		}
+
+		return uint64(time.Now().UnixNano()) / uint64(slotDuration.Nanoseconds()), nil
```

Comment:
might be useful to have a function to calculate slot from time. That could be reused here  and in GetEpochFromTime. 
Do we already have such a function though?

Reply:
is might be useful only for testing but for production use we never need to get the slot from real time